dbt-adapters/src/dbt/include/global_project/macros/materializations/snapshots/helpers.sql

```diff
@@ -109,7 +109,9 @@
         left outer join snapshotted_data
             on {{ unique_key_join_on(strategy.unique_key, "snapshotted_data", "source_data") }}
             where {{ unique_key_is_null(strategy.unique_key, "snapshotted_data") }}
-            or ({{ unique_key_is_not_null(strategy.unique_key, "snapshotted_data") }} and ({{ strategy.row_changed }})
+            or ({{ unique_key_is_not_null(strategy.unique_key, "snapshotted_data") }} and (
+               {{ strategy.row_changed }} {%- if strategy.hard_deletes == 'new_record' -%} or snapshotted_data.{{ columns.dbt_is_deleted }} = 'True' {% endif %}
```

Comment:
`snapshotted_data.{{ columns.dbt_is_deleted }}` is a varchar column or boolean? Should this be True without quotes?

Reply:
@colin-rogers-dbt It is a varchar, so it is correct as written.